app/services/reports/facility_progress_service.rb

```diff
@@ -82,15 +83,86 @@ def dimension_combinations_for(indicator)
+        hsh[record.period] = if diabetes_enabled
+          record[:daily_registrations_htn_or_dm]
+        else
+          record[:daily_registrations_htn_only] + record[:daily_registrations_htn_and_dm]
+        end
+      end
+    end
+
+    def diagnosis_headers
+      {
+        hypertension: I18n.t("progress_tab.diagnoses.hypertension_only"),
+        diabetes: I18n.t("progress_tab.diagnoses.diabetes_only"),
+        hypertension_and_diabetes: I18n.t("progress_tab.diagnoses.hypertension_and_diabetes")
+      }
     end
 
-    memoize def daily_follow_ups_grouped_by_day
-      scope = Reports::DailyFollowUp.with_hypertension
-      scope = scope.or(Reports::DailyFollowUp.with_diabetes) if diabetes_enabled
-      scope.where(facility: facility).group_by_day(:visited_at, last: 30).count
+    memoize def daily_registrations_breakdown
+      records = Reports::FacilityDailyFollowUpAndRegistration.for_region(region).where("visit_date >= ?", DAYS_AGO.days.ago.to_date)
+      records.each_with_object({}) do |record, hsh|
+        hsh[record.period] = {
+          hypertension: {
+            all: record[:daily_registrations_htn_only],
+            male: record[:daily_registrations_htn_only_male],
+            female: record[:daily_registrations_htn_only_female],
+            transgender: record[:daily_registrations_htn_only_transgender]
+          },
+          diabetes: {
+            all: record[:daily_registrations_dm_only],
+            male: record[:daily_registrations_dm_only_male],
+            female: record[:daily_registrations_dm_only_female],
+            transgender: record[:daily_registrations_dm_only_transgender]
+          },
+          hypertension_and_diabetes: {
+            all: record[:daily_registrations_htn_and_dm_all],
+            male: record[:daily_registrations_htn_and_dm_male],
+            female: record[:daily_registrations_htn_and_dm_female],
+            transgender: record[:daily_registrations_htn_and_dm_transgender]
+          }
+        }
+      end
+    end
+
+    memoize def daily_follow_ups_breakdown
+      records = Reports::FacilityDailyFollowUpAndRegistration.for_region(region).where("visit_date >= ?", DAYS_AGO.days.ago.to_date)
+      records.each_with_object({}) do |record, hsh|
+        hsh[record.period] = {
+          hypertension: {
+            all: record[:daily_follow_ups_htn_only],
+            male: record[:daily_follow_ups_htn_only_male],
+            female: record[:daily_follow_ups_htn_only_female],
+            transgender: record[:daily_follow_ups_htn_only_transgender]
+          },
+          diabetes: {
+            all: record[:daily_follow_ups_dm_only],
+            male: record[:daily_follow_ups_dm_only_male],
+            female: record[:daily_follow_ups_dm_only_female],
+            transgender: record[:daily_follow_ups_dm_only_transgender]
+          },
+          hypertension_and_diabetes: {
+            all: record[:daily_follow_ups_htn_and_dm_all],
```

Comment:
`daily_follow_ups_htn_and_dm_all` should be replaced by `daily_follow_ups_htn_or_dm`?